pandas/core/internals/array_manager.py

```diff
@@ -363,11 +363,7 @@ def putmask(self: T, mask, new, align: bool = True) -> T:
         )
 
     def diff(self: T, n: int, axis: int) -> T:
-        if axis == 1:
-            # DataFrame only calls this for n=0, in which case performing it
-            # with axis=0 is equivalent
-            assert n == 0
-            axis = 0
+        assert self.ndim == 2 and axis == 0  # caller ensures
```

Comment:
Worth setting `axis: int = 0` in the signature then?

Reply:
i didnt consider that, thought about removing the keyword entirely but decided to hold off for the time being

Reply:
> removing the keyword 

Yeah this would be reasonable too but didn't know if a base class signature needed to be matched.